src/ethereum/frontier/state.py

```diff
@@ -293,6 +293,33 @@ def account_exists(state: State, address: Address) -> bool:
     return get_account_optional(state, address) is not None
 
 
+def account_has_code_or_nonce(state: State, address: Address) -> bool:
+    """
+    Checks if an account has non zero nonce or non empty code
+
+    Parameters
+    ----------
+    state:
+        The state
+    address:
+        Address of the account that needs to be checked.
+
+    Returns
+    -------
+    has_code_or_nonce : `bool`
+        True if if an account has non zero nonce or non empty code,
+        False otherwise.
+    """
+    account = get_account_optional(state, address)
+    if account is None:
+        return False
+    else:
+        if account.nonce != Uint(0) or account.code != b"":
+            return True
+        else:
+            return False
```

Comment:
```suggestion
    return account is not None and (account.nonce != Uint(0) or account.code != b"")
```

Or

```suggestion
if account is None:
    return False

return account.nonce != Uint(0) or account.code != b""
```

Reply:
found a simpler way to do this:
```python
# since we are checking non empty nonce or code, it doesn't matter 
# if `get_account` returns `EMPTY_ACCCONT` when account is not present in state. 
account = get_account(state, address)
return account.nonce != Uint(0) or account.code != b""
```